api-error-response_test.go

```diff
@@ -0,0 +1,279 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2015 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required bZy applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+// Tests validate the Error generator function for http response with error.
+func TestHttpRespToErrorResponse(t *testing.T) {
+	// 'formulateAPIErrorResponse' formulates ErrorResponse for given APIError.
+	// provides a encodable populated response values.
+	formulateAPIErrorResponse := func(err APIError, bucketName string) ErrorResponse {
+		var data = ErrorResponse{}
+		data.Code = err.Code
+		data.Message = err.Description
+
+		data.BucketName = bucketName
+		// TODO implement this in future
+		return data
+	}
+
+	// Encodes the response headers into XML format.
+	encodeErr := func(response interface{}) []byte {
+		var bytesBuffer bytes.Buffer
+		bytesBuffer.WriteString(xml.Header)
+		encode := xml.NewEncoder(&bytesBuffer)
+		encode.Encode(response)
+		return bytesBuffer.Bytes()
+	}
+
+	// `formulteAPIErrorResponse` Mocks XML error response from the server.
+	formulteAPIErrorResponse := func(APIErr APIError, bucketName string) *http.Response {
```

Comment:
Looks like you have a name conflict here,  you avoided that by introducing a typo. This would be incorrect call this 

'genHTTPErrorResponse' and 'genAPIErrorResponse' - rather than using words like 'formulate' a better word would be just 'create' or 'gen',  pick easier names.